js/adapt-contrib-hotgraphic.js

```diff
@@ -48,12 +48,142 @@ define(function(require) {
             this.renderState();
             this.$('.hotgraphic-widget').imageready(_.bind(function() {
                 this.setReadyStatus();
+                this.setupTooltips();
             }, this));
 
             this.setupEventListeners();
         },
 
-        // Used to check if the hotgraphic should reset on revisit
+        setupTooltips: function() {
+            var tooltipConfig = this.model.get('_tooltips');
+            if (!tooltipConfig || !tooltipConfig._isEnabled) return;
+
+            this.model.get('_items').forEach(function(item, index) {
+                var config = {
+                    tooltipConfig: tooltipConfig,
+                    tooltipElement: $(this.$('.hotgraphic-tooltip')[index]),
+                    pinElement: $(this.$('.hotgraphic-graphic-pin')[index]),
+                    item: item
+                };
+                this.setTooltipPosition(config);
+                this.setTooltipEventListener(config);
+            }, this);
+        },
+
+        setTooltipPosition: function(config) {
+            var directionOpposites = {
+                left: 'right',
+                top: 'bottom',
+                right: 'left',
+                bottom: 'top'
+            };
+            var alignedPosition = this.getTooltipAlignedPosition(config);
+            var oppositeAlignment = directionOpposites[config.tooltipConfig._alignment];
+
+            if (!this.checkTooltipWithinBounds(config, alignedPosition)) {
+                alignedPosition = this.getTooltipAlignedPosition(config, oppositeAlignment);
+            }
+
+            if (!this.checkTooltipWithinBounds(config, alignedPosition)) {
```

Comment:
This if condition is identical to the one above

Reply:
it checks a different value of `alignedPosition`